### DISCUSSION TITLE
Edit Traits of Map

### DISCUSSION THREAD

**Discussion by BrendanMcDonagh-ContactFunnels** [GrapesJS/grapesjs#4833] [General]
Hi,

I am very new to GrapesJS and am at the stage where I able to create a canvas and integrate the webpage preset plugin. 

I am trying to understand how I would go about giving default traits for a map when it is dragged on to the canvas. i.e how to give default address for the map being dragged on to the canvas.

Any help would be greatly appreciated.

Regards

**Comment by artf** [comment]
Hi @BrendanMcDonagh-ContactFunnels traits read data directly from the component so it case of the Map block ([here](https://github.com/artf/grapesjs-blocks-basic/blob/f7c9869e345106ad533a682fad4aafbae62264f1/src/blocks.ts#L285-L294) you can see how it's added from the basic blocks plugin) you can update the default content with something like this:
```js
editor.Blocks.get('map').getContent().address = 'my address';
```
The next time you'll drop the map block, the component will be created with those properties (and you will see them in traits).
